### PR TITLE
feat: add webhook - openapi

### PR DESCRIPTION
**Pull request by alli83**
| Q             | A
| ------------- | ---
| Branch?       | main
| Tickets       | 
| License       | MIT
| Doc PR        | 

Possible to configure a webhook via an attribute:

`
#[ApiResource(operations: [

    new Post(openapi: new Webhook(name: 'a cute name')
    )),
    new Get(openapi: new Webhook(
        name: 'a cute name',
        pathItem: new PathItem(
            get: new Operation(
                summary: 'Something else here',
                description: 'Something else here for example',
            ),
        )
    )),
]
)]`
and generate documentation

### CODE REVIEW

File: src/OpenApi/Factory/OpenApiFactory.php [hunk 1]
```diff
@@ -227,9 +240,9 @@ private function collectPaths(ApiResource $resource, ResourceMetadataCollection
                 $openapiOperation = $openapiOperation->withDeprecated((bool) $operation->getOpenapiContext()['deprecated']);
             }
 
-            if ($path) {
+            if (!$pathItem && $path) {
                 $pathItem = $paths->getPath($path) ?: new PathItem();
```
Comment (alli83) [inline src/OpenApi/Factory/OpenApiFactory.php:244]: I will update: just $path should be enough

File: src/Metadata/Get.php [hunk 2]
```diff
@@ -43,7 +44,7 @@ public function __construct(
         array $paginationViaCursor = null,
         array $hydraContext = null,
         array $openapiContext = null,
-        bool|OpenApiOperation $openapi = null,
+        bool|OpenApiOperation|Webhook $openapi = null,
```
Comment (soyuka) [inline src/Metadata/Get.php:47]: we should probably change every operation that inherits from HttpCollection no?

Comment (alli83) [reply src/Metadata/Get.php:47]: Yes, I will update it. Apologies, I just forgot.

**Comment by soyuka** [comment]
Nice! Do you think we could add a functional test to `features/openapi/docs.feature` with a webhook?

**Review by soyuka — APPROVED** [review]

File: tests/Fixtures/TestBundle/Entity/DummyWebhook.php [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+class DummyWebhook
+{
+    /**
+     * @var int|null The id
+     */
+    #[ORM\Column(type: 'integer', nullable: true)]
+    #[ORM\Id]
+    #[ORM\GeneratedValue(strategy: 'AUTO')]
+    private $id;
+
```
Comment (soyuka) [inline tests/Fixtures/TestBundle/Entity/DummyWebhook.php:52]:
```suggestion
```

**Pull request merged**